src/Calculator/Views/CalculatorProgrammerBitFlipPanel.xaml.cpp

```diff
@@ -190,10 +193,13 @@ void CalculatorProgrammerBitFlipPanel::UpdateCheckedStates(bool updateAutomation
     m_updatingCheckedStates = true;
     auto it = m_flipButtons.begin();
     int index = 0;
+    bool mustUpdateTextOfMostSignificantDigits = m_currentValueBitLength != Model->ValueBitLength;
+    int previousMSDPosition = GetIndexOfLastBit(m_currentValueBitLength);
+    int newMSDPosition = GetIndexOfLastBit(Model->ValueBitLength);
     for (bool val : Model->BinaryDigits)
     {
         FlipButtons ^ flipButton = *it;
-        if (updateAutomationPropertiesNames)
+        if (updateAutomationPropertiesNames || (mustUpdateTextOfMostSignificantDigits && (index == previousMSDPosition || index == newMSDPosition)))
```

Comment:
I am unsure if it is related to this PR, but I get the incorrect value being read out from Narrator in this scenario:

1. Set size to "BYTE"
2. Toggle the most significant bit to 1
3. Set size to "QWORD"
4. All of the previously disabled bits are now set to 1 **(Root cause of the bug?)**
5. Narrator reads them all as 0 instead of 1 as you tab to each one

Let me know if I should just write this up as a separate issue.

Reply:
This is the expected behavior. 
If the most significant bit == 1, it means the number is negative, and based on Two's complement, it's how you should represent a negative number: https://en.wikipedia.org/wiki/Two%27s_complement

**-2**
Byte: ‭11111110‬
QWORD: ‭1111111111111111111111111111111111111111111111111111111111111110‬

What is not normal is that the narrator says "0" instead of "1", it is not related to this PR, but I will include the fix to it

Reply:
Fixed!